### PR TITLE
Virtual tracking

### PR DESCRIPTION
**Pull request by anpolimus**
Logging functionality for Virtual Y. 
We've implemented it as a separate module that could be enabled/disabled anytime.

- [ ] Enable openy_gc_log module
- [ ] Log in to the system
- [ ] open different content at Virtual Y
- [ ] go to /admin/virtual-y-logs and verify that tracking of Virtual Y events works.

### CODE REVIEW

**Comment by AndreyMaximov** [comment]
retest

File: modules/openy_gc_log/src/Form/LogEntityDeleteForm.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+namespace Drupal\openy_gc_log\Form;
+
+use Drupal\Core\Entity\ContentEntityDeleteForm;
+
+/**
+ * Provides a form for deleting Log entity entities.
+ *
+ * @ingroup openy_gc_log
+ */
+class LogEntityDeleteForm extends ContentEntityDeleteForm {
```
Comment (hamrant) [inline modules/openy_gc_log/src/Form/LogEntityDeleteForm.php:12]: Let's delete this and use ContentEntityDeleteForm in Entity.

File: openy_gated_content.info.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
-name: Open Y Gated Content base
+name: Vortual Y Base
```
Comment (hamrant) [inline openy_gated_content.info.yml:1]: Typo, we need **Virtual** here

File: modules/openy_gc_log/js/logger.behaviors.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+(function ($, Drupal) {
```
Comment (hamrant) [inline modules/openy_gc_log/js/logger.behaviors.js:1]: Why we need jquery for this? Can we make this api call with axios from the VUEX action?

**Comment by AndreyMaximov** [comment]
@anpolimus hi.
Please check http://ci.fivejars.com:8080/blue/organizations/jenkins/OPENY_GC_PR/detail/OPENY_GC_PR/229/pipeline for the PHPCS report

File: modules/openy_gc_log/openy_gc_log.info.yml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+name: Open Y Virtual YMCA Log
+type: module
+description: 'Virtual Y Log'
+package: Open Y Virtual YMCA
+version: 0.1
+core: 8.x
```
Comment (podarok) [inline modules/openy_gc_log/openy_gc_log.info.yml:6]: add drupal 9 version constraints, please

File: modules/openy_gc_log/openy_gc_log.libraries.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+logger:
+  js:
```
Comment (podarok) [inline modules/openy_gc_log/openy_gc_log.libraries.yml:2]: version?

File: modules/openy_gc_log/openy_gc_log.libraries.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    js/logger.behaviors.js: {}
+  dependencies:
+    - core/jquery
+    - core/drupalSettings
+    - openy_system/axios
+
+datepicker:
+  js:
```
Comment (podarok) [inline modules/openy_gc_log/openy_gc_log.libraries.yml:10]: version?

File: modules/openy_gc_log/src/Entity/LogEntity.php [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+
+  /**
+   * @param $fields
+   */
+  public static function defineEmailField(array &$fields) {
+    $fields['email'] = BaseFieldDefinition::create('string')
+      ->setLabel(t('Email'))
+      ->setDescription(t('The name of the Log entity entity.'))
+      ->setSettings([
+        'max_length' => 50,
```
Comment (podarok) [inline modules/openy_gc_log/src/Entity/LogEntity.php:81]: probably longer? Emails could be longer

File: openy_gated_content.info.yml [hunk 8]
```diff
@@ -1,8 +1,8 @@
-name: Open Y Gated Content base
+name: Virtual Y Base
 type: module
-description: 'Provides Open Y Gated Content functionality.'
-package: Open Y Gated Content
-version: 0.1
+description: 'Provides Open Y Virtual Y base module.'
+package: Open Y Virtual YMCA
+version: 0.2
```
Comment (podarok) [inline openy_gated_content.info.yml:5]: + Drupal 9 version constraints, please

**Pull request merged**